pkg/cmd/backuprestore/backupserver.go

```diff
@@ -61,7 +86,49 @@ func (b *backupServer) Run(ctx context.Context) error {
 	defer cancel()
 	signal.NotifyContext(cCtx, shutdownSignals...)
 
+	if b.enabled {
+		schedule, err := cron.ParseStandard(b.schedule)
+		if err != nil {
+			pErr := fmt.Errorf("error parsing backup schedule %v: %w", b.schedule, err)
+			klog.Error(pErr)
+			return pErr
+		}
+
+		err = b.scheduleBackup(cCtx, schedule)
+		if err != nil {
+			return err
+		}
+	}
+
 	<-ctx.Done()
+	return nil
+}
+
+func (b *backupServer) scheduleBackup(ctx context.Context, schedule cron.Schedule) error {
```

Comment:
before I forget, we need some sensible tests for the whole command as well :)

Reply:
:D :D .. will start with it now 👍🏽

Reply:
I added tests in [Backupserver_test](https://github.com/openshift/cluster-etcd-operator/pull/1306/files#diff-97951c69d5abc4a441c76a64b73793e40d36d8bf53c80d73f175db1231be3b3d )

I tested only the validate() for the following reasons 

- The backup tests are already well tested, I embedded the `backupOptions`
- The `Run()` is nothing more than a ticker, that takes the backup every tick
- The only thing I could test is the validation upon various input 

Please correct me if i am wrong

Reply:
I made various changes to facilitate testing 

I add a interface for `runBackup()`, and add a mock within the test to control the output 

However, I cant assert upon the tickers !! 

ptal

Reply:
> However, I cant assert upon the tickers !!

why not? just record the number of invocations in your mock.

Reply:
lgtm 👍🏽

Reply:
worked 👍🏽 thxxxx